src/subclasses/ContractInteractions.ts

```diff
@@ -440,26 +437,27 @@ class ContractInteractions {
     const offsetHelper = this.getOffsetHelperContract(signer);
 
     const offsetTxn: ContractTransaction =
-      await offsetHelper.autoOffsetUsingPoolToken(this.addresses.nct, amount, {
+      await offsetHelper.autoOffsetPoolToken(this.addresses.nct, amount, {
         gasLimit: GAS_LIMIT,
       });
     return await offsetTxn.wait();
   };
 
   /**
    *
-   * @description swaps given token for carbon pool tokens and uses them to retire carbon
-   * @notice this method may take up to even 1 minute to give a result
+   * @description Retire a specified amount of  carbon credits using the lowest quality (oldest) TCO2 tokens available from the specified Toucan token pool by sending ERC20
+   * tokens (cUSD, USDC, WETH, WMATIC).
+   * @notice This method needs two different actions signed and may take up to even 1 minute to give a result
+   * @param swapToken portal for the token to swap into pool tokens (only accepts WETH, WMATIC and USDC)
    * @param pool symbol of the pool (token) to use
    * @param amount amount of CO2 tons to offset
-   * @param swapToken portal for the token to swap into pool tokens (only accepts WETH, WMATIC and USDC)
    * @param signer this being a write transaction, we need a signer
    * @returns offset transaction
    */
-  autoOffsetUsingSwapToken = async (
+  autoOffsetExactOutToken = async (
```

Comment:
I don't understand the function name. Can you explain?

Reply:
Retiring carbon tokens requires multiple steps and interactions with Toucan Protocol's main contracts:

- Obtain a Toucan pool token such as BCT or NCT (by performing a token swap).
- Redeem the pool token for a TCO2 token.
- Retire the TCO2 token.

These steps are combined in each of the following "auto offset" methods implemented in OffsetHelper to allow a retirement within one transaction:

`autoOffsetPoolToken()` if the user already owns a Toucan pool token such as BCT or NCT,
`autoOffsetExactOutETH()` if the user would like to perform a retirement using MATIC, specifying the exact amount of TCO2s to retire,
`autoOffsetExactInETH()` if the user would like to perform a retirement using MATIC, swapping all sent MATIC into TCO2s,
`autoOffsetExactOutToken()` if the user would like to perform a retirement using an ERC20 token (USDC, WETH or WMATIC), specifying the exact amount of TCO2s to retire,
`autoOffsetExactInToken()` if the user would like to perform a retirement using an ERC20 token (USDC, WETH or WMATIC), specifying the exact amount of token to swap into TCO2s.

Reply:
@Niwilai the function names relate to swap functions that have the same naming. In both cases you need to swap. 
The difference is that for `OutETH` means you are specifying the exact amount of TCO2s to retire. for this function you will have to call `calculateNeededETHAmount` to know how many tokens you need to send. 

In means that you define the amount of tokens you want to swap for pool tokens. Here the amount you want to pay is more important than the amount of TCO2 you offset. useful for a subscription where you would want to retire 5$ per month. here you can use the `calculateExpectedPoolTokenForETH` to calculate beforehand how many TCO2 s you will get back/ you can retire.